http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/KubernetesServiceInterp.scala

```diff
@@ -97,8 +101,11 @@ class LeoKubernetesServiceInterp[F[_]: Parallel](
       else F.unit
 
       clusterId = saveClusterResult.minimalCluster.id
-      saveNodepool <- F.fromEither(getUserNodepool(clusterId, userInfo, req, ctx.now))
-      nodepool <- nodepoolQuery.saveForCluster(saveNodepool).transaction
+      claimedNodepoolOpt <- nodepoolQuery.claimNodepool(clusterId).transaction
```

Comment:
So this should transactionally claim the nodepool -- I checked and our isolation level is `RepeatableRead` which I think should be good. 👍